### PR TITLE
Run TestAOTAutograd with dynamo

### PR DESCRIPTION
**Pull request by jamesjwu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127618
* #126999
* #126791
* __->__ #128047

My goal is to run these tests with the autograd cache on, but first I want them running with dynamo. These tests already caught an interesting issue so I thought it would be helpful to just have them. 

Next up I'll have a second subclass of these tests, run them twice, and expect a cache hit the second time from autograd.

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
okey dokey. Is it possible to make some of these xfails instead of skips?

**Comment by jamesjwu** [comment]
Yup, will xfail the ones that I think are known issues we should fix; don't think we need to xfail ones that just happen to fail now because the test just doesn't even make sense with a dynamo input (like, yes they're expected to fail, but even if they passed it would kinda be not useful information).

File: test/functorch/test_aotdispatch.py [hunk 1]
```diff
@@ -678,8 +722,6 @@ def f(a):
 
         out_ref = f(ref_view)
         out_test = f_compiled(test_view)
-        print(ref)
```
Comment (jamesjwu) [inline test/functorch/test_aotdispatch.py:681]: These probably shouldn't be here

**Comment by jamesjwu** [comment]
@pytochbot merge

**Pull request closed without merging**